acoustid/future/fpindex/updater.py

```
@@ -0,0 +1,329 @@
+                    loaded_count,
+                    total_count,
+                    (loaded_count / total_count * 100),
+                )
+                if shutdown_event.is_set():
+                    logger.info("Interrupting initial data load due to shutdown event")
+                    raise ShutdownError
+
+    logger.info("Initial data load complete. Processed %d fingerprints.", loaded_count)
+
+
+async def replicate_data(
```

Ruff (0.8.2)

`replicate_data` is too complex (22 > 10)

(C901)